src/Login/ProtectedContent.jsx

```diff
@@ -70,6 +72,10 @@ class ProtectedContent extends React.Component {
                 const latestState = { ...newState };
                 latestState.dataLoaded = true;
                 this.setState(latestState);
+                if (newState.authenticated && isEnabled('workspaceTokenServiceRefreshTokenAtLogin')) {
+                    // initialize WTS:
+                    initWorkspaceRefreshToken();
```

Comment:
@paulineribeyre would that be a problem? I think it is a really light call.

Reply:
Yes i think that's a problem, from a quick search i see 26 `ProtectedContents` in `src/index.jsx`, we should not make so many API calls every time someone loads the page. Can we make a single call to WTS when we portal is loaded?

Reply:
Interesting, since the same could be said regarding the call to `checkLoginStatus` just a few lines up:

```js
.then(
            () => this.checkLoginStatus(store, this.state)
```
Maybe that is why `checkLoginStatus` has [this code ](https://github.com/uc-cdis/data-portal/pull/1095/files#diff-c8ea73aaef1661450270bc0f167f41abd12c3d3a78f9894a48460f3727aeb0baR123-R126) to avoid repeated calls? Should we do something similar here? Or should we try to find another location for our init code?
```js
  checkLoginStatus = (store, initialState) => {
    const newState = { ...initialState };
    const nowMs = Date.now();
    newState.authenticated = true;
    newState.redirectTo = null;
    newState.user = store.getState().user;
    if (nowMs - lastAuthMs < 60000) {
      // assume we're still logged in after 1 minute ...
      return Promise.resolve(newState);
    }
  ...
```
@mfshao what would you suggest for this?

Reply:
you can do something similar like this

or change in your wts init code so that if a user is not logged in, don't force them to do an OIDC dance with WTS

Reply:
i am curious why you chose to add it to `ProtectedContent` instead of `index.jsx` like in your initial code?

Reply:
@paulineribeyre I was asked by @mfshao to do this...I now realise this was in slack and is not here in the PR comments. 

@mfshao do you mind replying to this here?

Reply:
@paulineribeyre @mfshao I added a proposal here in this last commit: https://github.com/uc-cdis/data-portal/pull/1095/commits/82b7b20dd878e31ad3137ba41619ed758e315f65. Please let me know what you think. It is based on the current solution for `checkLoginStatus` in `ProtectedContent.jsx`